### PR TITLE
Ui viewer

### PR DESCRIPTION
**Pull request by spireken**

### CODE REVIEW

**Comment by spireken** [comment]
I debated using something like the Prototype pattern to dynamically instantiate the test widgets, but with the current requirements I kept it simple and straightforward.

File: Applications/Spire/Include/Spire/UiViewer/ColorSelectorButtonTestWidget.hpp [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+  //! Displays a widget for testing a ColorSelectorButton.
+  class ColorSelectorButtonTestWidget : public QWidget {
+    public:
+
+      //! Constructs a ColorSelectorButtonTestWidget.
+      /*
+        \param parent The parent widget.
+      */
+      ColorSelectorButtonTestWidget(QWidget* parent = nullptr);
```
Comment (kranar) [inline Applications/Spire/Include/Spire/UiViewer/ColorSelectorButtonTestWidget.hpp:17]: Explicit.

File: Applications/Spire/Include/Spire/UiViewer/UiViewer.hpp [hunk 2]
```diff
@@ -2,6 +2,7 @@
 #define SPIRE_UI_VIEWER_HPP
 
 namespace Spire {
+  class UiViewerWindow;
```
Comment (kranar) [inline Applications/Spire/Include/Spire/UiViewer/UiViewer.hpp:5]: Forward declare all classes, helps autocompletion in IDE's.

File: Applications/Spire/Include/Spire/UiViewer/UiViewerWindow.hpp [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+  //! Displays a window to test Spire widgets.
+  class UiViewerWindow : public Window {
+    public:
+
+      //! Constructs a UiViewerWindow.
+      /*
+        \param parent The parent widget.
+      */
+      UiViewerWindow(QWidget* parent = nullptr);
```
Comment (kranar) [inline Applications/Spire/Include/Spire/UiViewer/UiViewerWindow.hpp:18]: Explicit.

**Review by kranar — CHANGES_REQUESTED** [review]
Huge improvement.

**Review by kranar — APPROVED** [review]

**Pull request merged**